app/models/notification.rb

```diff
@@ -411,6 +411,7 @@ def send_email
 #  post_number       :integer
 #  post_action_id    :integer
 #  high_priority     :boolean          default(FALSE), not null
+#  parent_id         :integer
```

Comment:
I wonder if we shouldn't have a type to?

- target_id
- target_type

Feels more inlined with what we have been doing at other places.

Reply:
Yes I also thought of the join table, and saw that tgx mentioned in internally. It's probably worth looking if it would work correctly for us @AndrewPrigorshnev

Reply:
Speaking of the type field, the table already has it:

https://github.com/discourse/discourse/blob/773697df6a082184dab0731083fff59ac3aa7453/app/models/notification.rb#L404

It's used for the association here:

https://github.com/discourse/discourse/blob/773697df6a082184dab0731083fff59ac3aa7453/plugins/chat/app/models/chat/mention.rb#L11

> The column name is a bit confusing to me, parent suggests that it is selecting from the same table to me (ie. subcategory has a parent category). Target feels feels more appropriate if we are avoiding adding a join table.

Agree with this, `target_id` seems much better.

> My initial thinking here is why don't we use a join table but it will likely need to be done in core

> Yes I also thought of the join table

The join table is another possible way of implementing this, yes. But my opinion is that we can benefit a lot from having this column on the notification object. I think this is the best way of modelling what we have in reality in our domain. There are many objects that may have several related notifications, one notification per each interested user. For example, a post may have N notifications, in the same way other objects may have N related notifications, for example:
- topic
- chat message
- chat channel
- mention
- chat mention
- poll
- etc.

But let's discuss.

Reply:
> I wonder if we shouldn't have a type too?
> 
> * target_id
> * target_type

> Speaking of the type field, the table already has it:
> https://github.com/discourse/discourse/blob/773697df6a082184dab0731083fff59ac3aa7453/app/models/notification.rb#L404

Hm, but actually we have a lot of notification types, and there can be several types of notifications related to the same type of target. So, yes, we need to add a `target_type` column if we go with this solution, we can't just use the `notification_type` column.